addons/mail/static/src/core/common/thread.js

```diff
@@ -79,6 +80,25 @@ export class Thread extends Component {
         this.messageHighlight = this.env.messageHighlight
             ? useState(this.env.messageHighlight)
             : null;
+        let alreadyScrolling = false;
+        this.messageRefs = reactive(
+            {
+                byId: new Map(),
+                highlight: this.messageHighlight,
```

Comment:
I suppose you want `highlight` there so it calls this reactive callback.

Don't you want to observe all uses of messageHighlight? Probably would be best to have `useEffect` on this one.
The callback can be a function so it's shared by reactive callback and this `useEffect`.

Doing so would keep `messageRefs` as simply a reactive map.